File: instantsearch/src/main/java/com/algolia/instantsearch/model/Facet.java
package com.algolia.instantsearch.model;

public class Facet {
    final public String value;
    public int count;
    public boolean isEnabled;

    public Facet(String value, int count, boolean isEnabled) {
        this.value = value;
        this.count = count;
        this.isEnabled = isEnabled;
    }

    public Facet(String value, int count) {
        this(value, count, false);
    }

    @Override
    public String toString() {
        return "Facet{" + "value='" + value + ", count=" + count + ", enabled=" + isEnabled + '}';
    }
}

File: instantsearch-core/src/commonMain/kotlin/com/algolia/instantsearch/filter/Facet.kt
package com.algolia.instantsearch.filter

import kotlinx.serialization.InternalSerializationApi
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
public data class Facet(
    /**
     * Name of the facet. Is equal to the value associated to an [Attribute].
     */
    @SerialName("value") val value: String,
    /**
     * Number of times this [value] occurs for a given [Attribute].
     */
    @SerialName("count") val count: Int,
    /**
     * Highlighted value.
     */
    @SerialName("highlighted") val highlightedOrNull: String? = null
) {

    val highlighted: String
        get() = highlightedOrNull!!
}

@OptIn(InternalSerializationApi::class)
public operator fun List<Facet>.get(value: String): Int {
    return find { it.value == value }!!.count
}

@OptIn(InternalSerializationApi::class)
public operator fun Map<Attribute, List<Facet>>.get(attribute: Attribute, value: String): Int {
    return getValue(attribute).find { it.value == value }!!.count
}
